plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2524,6 +2531,7 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
 
         return tableStatisticsCache.computeIfAbsent(
                 new IcebergTableHandle(
+                        trinoCatalogHandle,
```

Comment:
Very optional thought, but doing `originalHandle.getCatalog()` looks more symmetrical with the rest of the code here.